hometask8.py

```diff
@@ -0,0 +1,80 @@
+import re
+
+phonebook = {}
+
+name = input('Enter a name:')
+
+number = input('Enter a number:')
+
+if phonebook.get(name):
+    print(f'{name} already exists. You need to delete a record and add again if you want to change it')
+else:
+    phonebook[name] = number
+    ukrphones = ''.join(re.findall("([+]?380\d{9}|0\d{9})", number))
```

Comment:
Check for a number before you add it to a phone book. There shouldn't be non-ukrainian numbers in your phonebook

Reply:
And your regular expression accept such invalid cases as:
```
809512345678
3809512345678912
phone380951234567

Reply:
I am already a bit confused with the code but I think I have fixed the issues

Reply:
Your regex still allows these formats:
```
3809512345678912
phone380951234567

Reply:
There must be some mistake. My regex is "(^\W*[+]?380[\d]{9}|0[\d]{9}$)" and it doesn't allow such numbers